File: Lib/Telerik/Examples/QuickStart/TreeView/Customize/CustomNodes/CustomTreeNodeElement.vb
Imports System.Text
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.TreeView.Customize.CustomNodes
	Public Class CustomTreeNodeElement
		Inherits TreeNodeElement
		Protected Overrides Sub InitializeFields()
			MyBase.InitializeFields()
			Me.StretchHorizontally = False
		End Sub

		Protected Overrides Function CreateContentElement() As TreeNodeContentElement
			Return New CustomContentElement()
		End Function

		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(TreeNodeElement)
			End Get
		End Property
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/TreeView/Customize/CustomNodes/CustomTreeNodeElement.cs
using System;
using System.Collections.Generic;
using System.Text;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.TreeView.Customize.CustomNodes
{
    public class CustomTreeNodeElement : TreeNodeElement
    {
        protected override void InitializeFields()
        {
            base.InitializeFields();
            this.StretchHorizontally = false;
        }

        protected override TreeNodeContentElement CreateContentElement()
        {
            return new CustomContentElement();
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(TreeNodeElement);
            }
        }
    }
}
